### ISSUE TITLE
Cannot index arrays with unsigned types

### ISSUE THREAD

**Issue by dylon** [lcompilers/lpython#2348]
It may be by design, but lpython does not support indexing arrays with unsigned integers. For example, when I try to run this with lpython:

```python
from lpython import Array, u32

from numpy import empty, uint32

def main() -> None:
    xs: Array[u32, 32] = empty(32, dtype=uint32)

    i: u32
    for i in range(32):
        xs[i] = i * i

    print(xs[5])  #=> 25

main()
```

I get the following error:

```shell
$ ./src/bin/lpython /var/tmp/main.py
semantic error: Type mismatch in index, expected a single integer or slice
  --> /var/tmp/main.py:10:12
   |
10 |         xs[i] = i * i
   |            ^ type mismatch (found: 'u32', expected: 'i32' or slice)


Note: Please report unclear or confusing messages as bugs at
https://github.com/lcompilers/lpython/issues.
```

If I make the index an `i32`, it works as expected:

```python
from lpython import Array, i32, u32

from numpy import empty, uint32

def main() -> None:
    xs: Array[u32, 32] = empty(32, dtype=uint32)

    i: u32
    for i in range(32):
        xs[i32(i)] = i * i

    print(xs[5])  #=> 25

main()
```

```shell
$ ./src/bin/lpython /var/tmp/main.py
25
```

If this is by design, it would be helpful for lpython to state so in its message.

**Comment by certik** [comment]
Thank! Indeed, if it is by design (I think it is right now), LPython should give a nice error message, explaining this.
